flask_saml_sso/sso.py

```diff
@@ -124,23 +140,37 @@ def api_token():
 
     If no user is currently logged in, redirect to SSO flow and return here
     """
+    logger = _get_logger()
+    logger.debug('API-token called')
+
     if not flask.session.get(session.LOGGED_IN):
         qargs = parse.urlencode({
             'next': flask.url_for('sso.api_token')
         })
-        redirect = flask.redirect("{}?{}".format(
-            flask.url_for('sso.sso'), qargs))
-        return redirect
+        redirect_url = "{}?{}".format(flask.url_for('sso.sso'), qargs)
+
+        logger.info('User not logged in, redirecting to {}'.format(
+            redirect_url))
+        return flask.redirect(redirect_url)
 
     app = flask.current_app
     # Check if user is permitted to create API tokens
 
     if app.config.get('SAML_API_TOKEN_RESTRICT', False):
+        logger.info('Token restriction enabled')
+
         group = app.config['SAML_API_TOKEN_RESTRICT_ATTR']
         value = app.config['SAML_API_TOKEN_RESTRICT_VALUE']
-
         attrs = flask.session.get(session.SAML_ATTRIBUTES)
+
+        logger.debug(
+            'SAML_API_TOKEN_RESTRICT_ATTR: {}'.format(group))
+        logger.debug(
+            'SAML_API_TOKEN_RESTRICT_VALUE: {}'.format(value))
+        logger.debug('SAML Attributes: {}'.format(attrs))
+
         if not attrs.get(group) or value not in attrs.get(group):
+            logger.info('User not permitted to create API tokens')
```

Comment:
Should this be an error or warning?

Reply:
I think errors or warnings should be used to communicate unexpected behavior in the application. This is a scenario we specifically designed it to handle.

Reply:
> I think errors or warnings should be used to communicate unexpected behavior in the application. This is a scenario we specifically designed it to handle.

Indeed, but I think we should consider warning about it; someone may be trying to do something they shouldn't be doing.

Reply:
That's fair. Let's make it a warning then.